src/lib.rs

```diff
@@ -149,7 +149,11 @@ where
     E: Borrow<Event<'a>>,
     F: fmt::Write,
 {
-    let mut state = state.unwrap_or_default();
+    let mut event_by_event = true;
```

Comment:
Is this something that can be activated as part of the event handling and be made part of the state?

To me it seems strange that there is a special case based on the state being present or not, and I fear it works only because it's specifically made to make the 'event-by-event' test pass.

Leaving it off definitely leads to some unmet expectations in the test suite.